cmd/pulumi-test-language/l2resourceasset_test.go

```diff
@@ -310,6 +310,32 @@ func (h *L2ResourceAssetArchiveLanguageHost) Run(
 		return nil, fmt.Errorf("could not register resource: %w", err)
 	}
 
+	remoteass, err := resource.NewURIAsset(
+		"https://raw.githubusercontent.com/pulumi/pulumi/master/cmd/pulumi-test-language/testdata/l2-resource-asset-archive/test.txt",
```

Comment:
Obviously it's never ideal having a test make a network call due to flakiness risks, but I thought this might be the safest best -- if CI is running enough to run this test, then the repository must be clonable and GitHub must be up. Moreover, this file is linked to this test suite, so again, the chances of this test and that file coexisting consistently feels as high as we can aim for. Happy to take feedback on this though :pray:

Reply:
> then the repository must be clonable and GitHub must be up

Kinda. Those cloning a repo and doing a file download are slightly different systems internally, and one can go down and not the other.  But more than that networks are just unreliable in general, and this makes it a little more annoying to run this locally.

If this was just Go we could use something like https://pkg.go.dev/net/http/httptest, but that's obviously harder across languages, so maybe this is the best we can do?  Or maybe we could have the conformance tests set up a server for all languages that would run for the duration of the tests? That would obviously mean rewriting the URL during the tests though, so I don't know if it's feasible.

Reply:
I think this is fine. If we see flakyness with this we can invest in doing something like @tgummerer has suggested.